cmd/kubeadm/app/cmd/certs_test.go

```diff
@@ -243,26 +251,35 @@ func TestRunRenewCommands(t *testing.T) {
 			}
 
 			// exec renew
-			renewCmds := getRenewSubCommands(os.Stdout, tmpDir)
-			cmdtestutil.RunSubCommand(t, renewCmds, test.command, fmt.Sprintf("--cert-dir=%s", tmpDir))
-
-			// check the file is modified
-			for _, cert := range test.Certs {
-				file, err := os.Stat(filepath.Join(tmpDir, fmt.Sprintf("%s.crt", cert.BaseName)))
-				if err != nil {
-					t.Fatalf("couldn't get certificate %s: %v", cert.Name, err)
-				}
-				if ModTime[cert.Name] == file.ModTime() {
-					t.Errorf("certificate %s was not renewed as expected", cert.Name)
-				}
+			renewCmds := getRenewSubCommands(tmpDir)
+			args := fmt.Sprintf("--cert-dir=%s", tmpDir)
+			if len(test.Args) > 0 {
+				args = test.Args + " " + args
 			}
```

Comment:
pinning the --cert-dir flag for all test cases seems fine.

Reply:
sorry, I didn't get it, `--cert-dir` is added to all testcases but only `test.Args` is added conditionally. And here we cannot append `test.Args` to all tests since empty string "" will be interpreted to a args by cobra.

Reply:
no, i was just saying this seems fine.